docs/worker_tags.md

```diff
@@ -0,0 +1,171 @@
+     celeborn.tags.tagsExpr: 'env=production'
+   tags:
+     env=production:
+       - 'host1:1111'
+       - 'host2:2222'
+     env=staging:
+       - 'host3:3333'
+     region=us-east:
+       - 'host1:1111'
+       - 'host3:3333'
+     region=us-west:
+       - 'host2:2222'
+-  tenantId: tenant_01
+   level: TENANT
+   config:
+     celeborn.tags.preferClientTagsExpr: false
+     celeborn.tags.tagsExpr: 'env=production,region=us-east'
+   users:
+     - name: Jerry
+       config:
+         celeborn.tags.preferClientTagsExpr: true
+```
+
+#### Database Store Backend
+
+This backend reads [worker tags and configurations](#configuration) settings from a
+user-specified database. For more information on using the database store backend,
+refer to [database config service](../developers/configuration#database-config-service).
+
+Here is an example SQL of worker tags assignment and configuration:
+
+```sql
+# SYSTEM level configuration
+INSERT INTO `celeborn_cluster_system_config` ( `id`, `cluster_id`, `config_key`, `config_value`, `type`, `gmt_create`, `gmt_modify` )
+VALUES
+    ( 1, 1, 'celeborn.tags.preferClientTagsExpr', 'true', 'master', '2024-02-27 22:08:30', '2024-02-27 22:08:30' ),
+    ( 2, 1, 'celeborn.tags.tagsExpr', 'env=production', 'master', '2024-02-27 22:08:30', '2024-02-27 22:08:30' ),
+
+# TENANT/TENANT_USER level configuration
+INSERT INTO `celeborn_cluster_tenant_config` ( `id`, `cluster_id`, `tenant_id`, `level`, `name`, `config_key`, `config_value`, `type`, `gmt_create`, `gmt_modify` )
+VALUES
+    ( 1, 1, 'tenant_01', 'TENANT', '', 'celeborn.tags.preferClientTagsExpr', 'true', 'master', '2024-02-27 22:08:30', '2024-02-27 22:08:30' ),
+    ( 2, 1, 'tenant_01', 'TENANT', '', 'celeborn.tags.tagsExpr', 'env=production,region=us-east', 'master', '2024-02-27 22:08:30', '2024-02-27 22:08:30' ),
+    ( 3, 1, 'tenant_01', 'TENANT_USER', 'Jerry', 'celeborn.tags.preferClientTagsExpr', 'true', 'master', '2024-02-27 22:08:30', '2024-02-27 22:08:30' ),
+
+# Worker Tags Assignment
+INSERT INTO `celeborn_cluster_tags` ( `id`, `cluster_id`, `tag`, `worker_id`, `gmt_create`, `gmt_modify` )
+VALUES
+    ( 1, 1, 'env=production', 'host1:1111', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+    ( 2, 1, 'env=production', 'host2:2222', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+    ( 3, 1, 'env=staging',    'host3:3333', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+    ( 4, 1, 'region=us-east', 'host1:1111', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+    ( 5, 1, 'region=us-east', 'host3:3333', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+    ( 6, 1, 'region=us-west', 'host2:2222', '2023-08-26 22:08:30', '2023-08-26 22:08:30' ),
+```
+
+## FAQ
+
+#### - What happens if no worker matches the specified tagsExpr?
+If no worker matches the specified tags expression, no workers will be selected
+for the shuffle. Depending on application configurations, it can fall back to
+Spark Shuffle.
```

Comment:
can we define the worker tag in dynamic config service for user identifier?

So that, the tag would be transparent for users?